.github/workflows/tests.yml

```diff
@@ -16,7 +17,121 @@ jobs:
     name: Static Analysis
     steps:
       - name: Checkout
-        uses: actions/checkout@v2
+        # for security reasons we pin commit ids and not tags.
+        # actions/checkout@v3.0.2 -> 2541b1294d2704b0964813337f33b291d3f8596b
+        uses: actions/checkout@2541b1294d2704b0964813337f33b291d3f8596b
+        with:
+          fetch-depth: 0
 
-      - name: Run pre-commit
-        run: make test/pre-commit
+      # - name: Setup SSH to Private Modules
+      #   # for security reasons we pin commit ids and not tags.
+      #   # webfactory/ssh-agent@v5.0.4 -> fc49353b67b2b7c1e0e6a600572d01a69f2672dd
+      #   uses: webfactory/ssh-agent@fc49353b67b2b7c1e0e6a600572d01a69f2672dd
+      #   with:
+      #     ssh-private-key: |
+      #       ${{ secrets.DEPLOY_KEY__REPOSITORY_A }}
+      #       ${{ secrets.DEPLOY_KEY__REPOSITORY_B }}
+
+      - name: Set up Golang
+        # for security reasons we pin commit ids and not tags.
+        # actions/setup-go@v3.2.0 -> b22fbbc2921299758641fab08929b4ac52b32923
+        uses: actions/setup-go@b22fbbc2921299758641fab08929b4ac52b32923
+        with:
+          go-version: 1.17.6
+
+      - name: Set up Terramate
+        # for security reasons we pin commit ids and not tags.
+        # mineiros-io/terramate@v0.1.28 -> 8535f918c0905572d846d82aa3d0feee23388f98
+        run: go install github.com/mineiros-io/terramate/cmd/terramate@8535f918c0905572d846d82aa3d0feee23388f98
+        env:
+          GOPROXY: direct
+
+      - name: Set up Terradoc
+        # for security reasons we pin commit ids and not tags.
+        # mineiros-io/terradoc@main -> af1a7b3ae3635958adf5ee2f40e0c3e70fd0803a
+        run: go install github.com/mineiros-io/terradoc/cmd/terradoc@af1a7b3ae3635958adf5ee2f40e0c3e70fd0803a
+        env:
+          GOPROXY: direct
+
+      - name: Set up Terraform
+        # for security reasons we pin commit ids and not tags.
+        # hashicorp/setup-terraform@v2.0.0 -> 17d4c9b8043b238f6f35641cdd8433da1e6f3867
+        uses: hashicorp/setup-terraform@17d4c9b8043b238f6f35641cdd8433da1e6f3867
+        with:
+          terraform_version: 1.2.1
```

Comment:
Why are we using an older version here?

Reply:
https://github.com/mineiros-io/terraform-module-template/blob/553f36cb66dbebf343e8ad3217883c3a40a1d53a/.github/workflows/tests.yml#L61

we can update the module template

Reply:
I think we should. The tests we have should be enough to discover any regressions related to the TF version.

Reply:
we do not use any newer features and just require >= 1.0 ... so any terraform version should work.. ;)